src/taco-cli/cli/kit.ts

```diff
@@ -664,27 +666,26 @@ class Kit extends commands.TacoCommandBase {
         if (!semver.valid(version)) {
             return Q.reject(errorHelper.get(TacoErrorCodes.ErrorInvalidVersion, version, "cordova"));
         }
-        CordovaHelper.ensureCordovaVersionAcceptable(version);
 
-        var npmCommand: string = process.platform === "win32" ? "npm.cmd" : "npm";
-        var npmProcess: child_process.ChildProcess = child_process.spawn(npmCommand, ["view", "cordova", "versions"]);
-        npmProcess.on("error", function (error: Error): void {
-            throw errorHelper.get(TacoErrorCodes.ErrorReadingPackageVersions, "cordova");
-        });
+        CordovaHelper.ensureCordovaVersionAcceptable(version);
+        
+        Q({})
+            .then(function(): any {
+                return NpmHelper.view(["cordova", "versions"], ".");
+            })
+            .then(function(result: any) {
+                var versions: any[] = [];
+                if (result) {
+                    var cordovaVersion = Object.keys(result)[0];
+                    versions = result[cordovaVersion].versions;
+                }
 
-        npmProcess.stdout.on("data", function (data: any): void {
-            try {
-                // The versions are returned as stringified array of strings
-                var versions: string = data.toString();
-                if (versions.indexOf("'" + version + "'") !== -1) {
+                if (versions.indexOf(version) !== -1) {
                     deferred.resolve(version);
```

Comment:
Now that everything is promisey, we don't need to use a deferred any more, you can just `return NpmHelper.view(...).then(...).then(() => {... return version })`